python/triqs/gf/gf.py

```diff
@@ -530,11 +521,15 @@ def _combine_mesh_mul(l, r):
     def __mul__(self,y):
         if isinstance(y, Gf):
             # make a copy, but special treatment of the mesh in the Imtime case.
-            c = Gf(mesh = Gf._combine_mesh_mul(self._mesh, y.mesh),
-                   data = self._data.copy(), 
-                   indices = self._indices.copy(), 
-                   name = self.name)
-            c.__imul__impl(y)
+            if self.target_rank == 2 and y.target_rank == 2 and self.target_shape != y.target_shape:
+                c = Gf(mesh=Gf._combine_mesh_mul(self._mesh, y.mesh),
+                       target_shape=[self.target_shape[0], y.target_shape[1]])
+            else:
+                c = Gf(mesh = Gf._combine_mesh_mul(self._mesh, y.mesh),
+                    data = self._data.copy(), 
+                    indices = self._indices.copy(), 
+                    name = self.name)
```

Comment:
```suggestion
            c = Gf(mesh=Gf._combine_mesh_mul(self._mesh, y.mesh),
                   target_shape=[self.target_shape[0], y.target_shape[1]])
```
I don't think we should make a distinction here. Note that we have deprecated the use of `self._indices` and the name of the Gf should not propagate through multiplication.